Cargo.toml

```diff
@@ -174,7 +174,7 @@ opt-level = "z"
 
 [features]
 use-synonyms-search = ["thesaurus/static"]
-default = ["memory-cache"]
+default = ["memory-cache", "socks"]
```

Comment:
It's bad user experience to require people to compile their own version of the software just to enable one simple feature. The code in the project itself is fairly minimal. And you don't need to gate the whole proxy feature behind the flag, because `reqwest` supports http(s) proxies even without `socks` enabled.

Reply:
> It's bad user experience to require people to compile their own version of the software just to enable one simple feature. The code in the project itself is fairly minimal, so the benefits of removing it would be pretty much 0. And you don't need to gate the whole proxy feature behind the flag, because `reqwest` supports http(s) proxies even without `reqwest/socks` enabled.

Yes, true but I think if you think about security purposes and user customizability purposes having this feature compiled based is quite beneficial like if we don't have this compiled based then this code to parse remains in the app which means if a bad actor has access to the config file then they can manipulate it to use their malicious proxy address which is not great instead having the user to whether include this code or not can greatly reduce surface attack area like for example if the user does not compile the app with the proxy code then the bad actor even if they change it to their malicious address the app will never use it because it never has the code to. So this improves security by alot by reducing surface attack area. Also, security sometimes can be linked with privacy as well like if the server gets hacked by using this surface attack area this can also harm user privacy like the hacker could indeed have hold of sensitive data of the user not great again. Moreover, by having this as a compile only option improves user customizability too like they can include features they want otherwise they can exclude it during build time. So, that's why I would suggest having it as a compile/build only option. I hope you got the idea 🙂 .

Reply:
> It's bad user experience to require people to compile their own version of the software just to enable one simple feature. The code in the project itself is fairly minimal, so the benefits of removing it would be pretty much 0. And you don't need to gate the whole proxy feature behind the flag, because `reqwest` supports http(s) proxies even without `reqwest/socks` enabled.

After reconsidering your suggestion, I think you are right we should not block the user for such a small feature. So in accordance to your suggestions I will make a few new suggestions for the changes to make I hope you don't mind it 🙂 .